test/setup_bundler_test.rb

```diff
@@ -5,18 +5,22 @@
 
 class SetupBundlerTest < Minitest::Test
   def test_does_nothing_when_running_in_the_ruby_lsp
+    Object.any_instance.expects(:system).with(bundle_install_command)
     run_script
     refute_path_exists(".ruby-lsp")
   end
 
   def test_does_nothing_if_both_ruby_lsp_and_debug_are_in_the_bundle
+    Pathname.any_instance.expects(:basename).returns("fake_project")
```

Comment:
Is there a risk that globally stubbing `Pathname` here could have unintentional side-effects, e.g. on logging during tests?

Reply:
I don't think so. This test only runs the `setup_bundler.rb` script and the only place we use `basename` is in the specific line we're testing.

Do you see another place where it could accidentally stub the wrong return?

Reply:
I'm thinking that a logging library or something else that's used in tests might rely on `basename` internally.

It's probably fine here, but in large systems I've seen problems caused by `any_instance` so I'm generally cautious of it.

(it's interesting to note that Minitest has [rejected](https://github.com/minitest/minitest/pull/245#issuecomment-148219318) this feature, it's only in Mocha).

Reply:
I tried swapping for `File.basename` but that is used by Minitest itself and stubbing it will break.

I think it should be fine for now. The only way to avoid the stub would be re-writing this to be a class to that it's not invoked in a script style anymore.